riptide-core/src/main/java/org/zalando/riptide/DefaultRequestArguments.java

```diff
@@ -21,7 +27,7 @@
     HttpMethod method;
 
     @Wither
-    URI baseUrl;
+    Supplier<URI> baseUrlProvider;
```

Comment:
This is also a breaking change if `DefaultRequestArguments` is really considered public and not for library-internal use only